xarray/core/dataset.py

```diff
@@ -874,6 +887,9 @@ def selkeys(dict_, keys):
             return dict((d, dict_[d]) for d in keys if d in dict_)
 
         def maybe_chunk(name, var, chunks):
+            if name in self.dims:
```

Comment:
Actually, maybe put this logic in `IndexVariable` instead? We could define a `chunk` method that looks like:
```
def chunk(self, ...):
    return self.copy(deep=False)
```